src/PostgREST/Error.hs

```diff
@@ -180,6 +181,11 @@ instance JSON.ToJSON ApiRequestError where
     "message" .= ("Could not choose the best candidate function between: " <> T.intercalate ", " [pdSchema p <> "." <> pdName p <> "(" <> T.intercalate ", " [ppName a <> " => " <> ppType a | a <- pdParams p] <> ")" | p <- procs]),
     "details" .= JSON.Null,
     "hint"    .= ("Try renaming the parameters or the function itself in the database so function overloading can be resolved" :: Text)]
+  toJSON (ColumnNotFound colName) = JSON.object [
+    "code"    .= ApiRequestErrorCode18,
+    "message" .= ("Could not find '" <> colName <> "' in the target table" :: Text),
+    "details" .= JSON.Null,
+    "hint"    .= ("If a new column was created in the database with this name, try reloading the schema cache." :: Text)]
```

Comment:
Should we include on `details` the list of detected columns? Similarly to how we do it for an error on a relationship?

https://github.com/PostgREST/postgrest/blob/b2935ef7990107642e261b25b6f30a2aed932864/test/spec/Feature/Query/EmbedDisambiguationSpec.hs#L17-L44

Reply:
I like it! Can't see a security problem either, the columns of tables/views exposed through PostgREST are already published in other ways.

Reply:
A more useful `hint` for this one. 

When referring to a bad column name, postgres tries to detect a typo:

```
$ curl 'localhost:3000/projects?select=id,nam'
{"code":"42703","details":null,"hint":"Perhaps you meant to reference the column \"projects.name\".","message":"column projects.nam does not exist"}

$ curl 'localhost:3000/projects?select=id,namee'
{"code":"42703","details":null,"hint":"Perhaps you meant to reference the column \"projects.name\".","message":"column projects.namee does not exist"}

$ curl 'localhost:3000/projects?select=id,nae'
{"code":"42703","details":null,"hint":"Perhaps you meant to reference the column \"projects.name\".","message":"column projects.nae does not exist"}

$ curl 'localhost:3000/projects?select=id,xxname'
{"code":"42703","details":null,"hint":"Perhaps you meant to reference the column \"projects.name\".","message":"column projects.xxname does not exist"}

## here it no longer suggest a column
$ curl 'localhost:3000/projects?select=id,yyxxname'
{"code":"42703","details":null,"hint":null,"message":"column projects.yyxxname does not exist"}
```

PostgreSQL does some fuzzy logic for this: [errorMissingColumn](https://github.com/postgres/postgres/blob/5035c93c8a5ac6804da79c67403460348b381924/src/backend/parser/parse_relation.c#L3625-L3696).

Perhaps we can use https://hackage.haskell.org/package/fuzzyfind-3.0.0/docs/Text-FuzzyFind.html for something similar?

Reply:
Oh yeah, nice, quite helpful indeed! Save that work for another PR down the road?

Reply:
Done and pushed.

Reply:
Sorry for the late reply here.

This is something that still worries me, I've got many users at Supabase that require hand-holding and removing this error is no good for them.

Could you try to add a hint for columns with similar names? On https://github.com/PostgREST/postgrest/pull/2575 we're using the `FuzzySet` library, that might help you get started.

Reply:
Sorry, I just noticed this: are we talking about errors when using unknown columns with `?select`? This PR only changes the error for unknown `?columns`. The behaviour for `select` remains the same:

```
curl 'localhost:3000/todos?select=id,xdone'
# before
{"code":"42703","details":null,"hint":"Perhaps you meant to reference the column \"todos.done\".","message":"column todos.xdone does not exist"}
# after
{"code":"42703","details":null,"hint":"Perhaps you meant to reference the column \"todos.done\".","message":"column todos.xdone does not exist"}
```

This PR only changes the message for bad `columns`:

```
curl --request POST \
  --url 'http://localhost:3000/todos?columns=xtask' \
  --header 'Content-Type: application/json' \
  --header 'Prefer: return=representation' \
  --data '{"id":5, "task": "buy eggs", color": "#001100"}'
# before
{"code":"42703","details":null,"hint":null,"message":"column \"xtask\" of relation \"todos\" does not exist"}
# after
{"code":"PGRST118","details":null,"hint":null,"message":"Column 'xtask' does not exist"}
```

Indeed the table name was there before, but other than that pretty much the same, no fuzzy suggestions from PostgreSQL in this case.

So are we okay in that case? I certainly don't want to add to your support burden and I feel like this PR should not since the behaviour is the same before and after.

Reply:
Oh, I see. I was thinking that PostgreSQL suggestions for `SELECT` would apply to insertions as well.

The table name on the error message would still be nicer but not critical as for now we can only do an insert on a single table at a time.

> So are we okay in that case?

Yes, it's all good now.

Reply:
Table name already added! Agreed, more of a nice to have but it's more consistent with the select error.